gems/smithy/lib/smithy/anvil/client/request_syntax_example.rb

```diff
@@ -0,0 +1,162 @@
+
+              output = client.#{@method_name}#{request_params}
+          EXAMPLE
+          example.split("\n")
+        end
+
+        private
+
+        def request_params
+          target = @operation['input']['target']
+          return '' if target == 'smithy.api#Unit'
+
+          input = @shapes[target]
+          return '' unless input['members'].any?
+
+          "(\n#{struct(input, '  ', Set.new, top_level: true)}\n  )"
+        end
+
+        # rubocop:disable Metrics
+        def value(target, indent, visited)
+          if visited_shape?(target, visited)
+            shape = Vise::Shape.name(target)
+            return "{\n#{indent}  # recursive #{shape}\n#{indent}}"
+          end
+
+          visited += [target]
+          shape = @shapes[target]
+          type_or_target = (shape['type'] if shape) || target
+
+          case type_or_target
+          when 'blob', 'smithy.api#Blob' then blob(shape)
+          when 'boolean', 'smithy.api#Boolean' then 'false'
+          when 'string', 'smithy.api#String' then string(target)
+          when 'byte', 'smithy.api#Byte' then '97'
+          when 'short', 'smithy.api#Short', 'integer', 'smithy.api#Integer',
+               'long', 'smithy.api#Long', 'bigInteger', 'smithy.api#BigInteger' then '1'
+          when 'float', 'smithy.api#Float', 'double', 'smithy.api#Double' then '1.0'
+          when 'bigDecimal', 'smithy.api#BigDecimal' then 'BigDecimal(1)'
+          when 'timestamp', 'smithy.api#Timestamp' then 'Time.now'
+          when 'document', 'smithy.api#Document' then 'TODO: document'
+          when 'enum' then enum(shape)
+          when 'intEnum' then enum(shape, string: false)
+          when 'list' then list(shape, indent, visited)
+          when 'map' then map(shape, indent, visited)
+          when 'structure' then struct(shape, indent, visited)
+          when 'union' then 'TODO: union'
+          else raise "unsupported type or target: #{type_or_target}"
+          end
+        end
+        # rubocop:enable Metrics
+
+        def blob(shape)
+          if shape && shape['traits']&.include?('smithy.api#streaming')
+            'File.read("source_file")'
+          else
+            '"data"'
+          end
+        end
+
+        def string(target)
+          Vise::Shape.name(target).inspect
+        end
+
+        def enum(shape, string: true)
+          enum_values = shape['members'].map do |_, member_shape|
+            value = member_shape['traits']['smithy.api#enumValue']
+            string ? "\"#{value}\"" : value
+          end
+          "#{enum_values.first} # One of: [#{enum_values.join(', ')}]"
+        end
+
+        def visited_shape?(target, visited)
+          # Prelude shapes are used more than once usually
+          !Vise::PRELUDE_SHAPES.include?(target) && visited.include?(target)
+        end
+
+        def struct(struct_shape, indent, visited, top_level: false)
+          lines = []
+          lines << '{' unless top_level
```

Comment:
we're adding {} at the top level as well - could we remove the special case for top_level here?

Reply:
My latest commit removes top level brackets. Which do you prefer? Outer hash causes another level of indent but outer hash is necessary for options (not shown here)

Reply:
I'd lean towards not using the outer hashes I think, so ignore this :-)

Reply:
I may change this to multiline:
```
params = {
 ...
}
options = { ... }
operation(params, options)
```